FinSearchUnified/Helper/StaticHelper.php

```diff
@@ -427,4 +427,33 @@ public static function getShopwareVersion()
 
         return ltrim($version, 'v');
     }
+
+    public static function getPreferredImage(
+        $image,
+        $thumbnails,
+        $imageSize,
+        $preferredExportWidth,
+        $preferredExportHeight
+    ) {
+        $preferredExportResolution = sprintf("%dx%d", $preferredExportWidth, $preferredExportHeight);
+        if (array_key_exists($preferredExportResolution, $thumbnails)) {
+            $image = $thumbnails[$preferredExportResolution];
+        } elseif ($imageSize < $preferredExportWidth * $preferredExportHeight) {
+            foreach (array_keys($thumbnails) as $resolution) {
+                $dimensions = explode('x', $resolution);
+                $thumbnailsSizes[$resolution] = intval($dimensions[0]) * intval($dimensions[1]);
+            }
+            ksort($thumbnailsSizes, SORT_NUMERIC);
+            $index = 0;
+            foreach ($thumbnailsSizes as $key => $thumbnailsSize) {
+                if ($thumbnailsSize > $imageSize) {
+                    $index = $key;
+                    break;
+                }
+            }
+            $image = $thumbnails[$index];
+        }
+
+        return $image;
```

Comment:
This code looks too much stuck together. And it can be a bit simplified.

```suggestion
        $preferredExportResolution = sprintf("%dx%d", $preferredExportWidth, $preferredExportHeight);

        if (array_key_exists($preferredExportResolution, $thumbnails)) {
            return $thumbnails[$preferredExportResolution];
        } elseif ($imageSize < $preferredExportWidth * $preferredExportHeight) {
            $thumbnailsSizes = [];
            foreach (array_keys($thumbnails) as $resolution) {
                $dimensions = explode('x', $resolution);
                $thumbnailsSizes[$resolution] = intval($dimensions[0]) * intval($dimensions[1]);
            }
            
            ksort($thumbnailsSizes, SORT_NUMERIC);

            foreach ($thumbnailsSizes as $key => $thumbnailsSize) {
                if ($thumbnailsSize > $imageSize) {
                    return $thumbnails[$key];
                }
            }
        }

        return $image;
```